### PR TITLE
fix: migrate to flutter 2.5

### LINKED ISSUE #135: CI Failing due to usage of some deprecated features
**Describe the bug**
CI fails after #84 due to the following features ( `accentColor`, `accentTextTheme` and `brightness` )
These errors are thrown by `flutter analyze` and the log of it is attached below :

<details>
<summary>CI Flutter analyze log</summary>

```
Analyzing mobile-app...                                         

   info • 'accentColor' is deprecated and shouldn't be used. Use colorScheme.secondary instead. For more information, consult the migration guide at https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties#migration-guide. This feature was deprecated after v2.3.0-0.1.pre. • lib/cv_theme.dart:56:9 • deprecated_member_use
   info • 'accentColor' is deprecated and shouldn't be used. No longer used by the framework, please remove any reference to it. For more information, consult the migration guide at https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties#migration-guide. This feature was deprecated after v2.3.0-0.1.pre. • lib/ib_theme.dart:11:7 • deprecated_member_use
   info • 'accentTextTheme' is deprecated and shouldn't be used. No longer used by the framework, please remove any reference to it. For more information, consult the migration guide at https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties#migration-guide. This feature was deprecated after v2.3.0-0.1.pre. • lib/ib_theme.dart:21:7 • deprecated_member_use
   info • 'accentTextTheme' is deprecated and shouldn't be used. No longer used by the framework, please remove any reference to it. For more information, consult the migration guide at https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties#migration-guide. This feature was deprecated after v2.3.0-0.1.pre. • lib/ib_theme.dart:21:42 • deprecated_member_use
   info • 'accentColor' is deprecated and shouldn't be used. Use colorScheme.secondary instead. For more information, consult the migration guide at https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties#migration-guide. This feature was deprecated after v2.3.0-0.1.pre. • lib/main.dart:43:15 • deprecated_member_use
   info • 'accentColor' is deprecated and shouldn't be used. Use colorScheme.secondary instead. For more information, consult the migration guide at https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties#migration-guide. This feature was deprecated after v2.3.0-0.1.pre. • lib/main.dart:55:15 • deprecated_member_use
   info • 'brightness' is deprecated and shouldn't be used. This property is no longer used, please use systemOverlayStyle instead. This feature was deprecated after v2.4.0-0.0.pre. • lib/ui/views/ib/ib_landing_view.dart:71:7 • deprecated_member_use

7 issues found. (ran in 20.7s)
```
</details>

Checklist :
- [ ] `accentColor` depreciated, replace it with `colorScheme.secondary`
- [ ] `accentTextTheme` depreciated, replace it with `colorScheme.secondary`
- [ ] `brightness` depreciated, replace it with `ThemeData.estimateBrightnessForColor()`

More info: https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties#accentcolorbrightness
Welcome to Flutter 2.5 💯 

_edited by @manjotsidhu_

### LINKED ISSUE #138: [BUG] ERROR: The argument type 'PointerEvent' can't be assigned to the parameter type 'PointerDownEvent'.
**Describe the bug**
After updating `Flutter (Channel stable, 2.5.0)` to the latest versions, I'm getting the following error when building:

/../AppData/Local/Pub/Cache/hosted/pub.dartlang.org/photo_view-0.11.1/lib/src/core/photo_view_gesture_detector.dart:106:29: Error: The argument type '`PointerEvent`' can't be assigned to the parameter type '`PointerDownEvent`'.

    '`PointerEvent`' is from 'package:flutter/src/gestures/events.dart' ('/../flutter/packages/flutter/lib/src/gestures/events.dart').
    '`PointerDownEvent`' is from 'package:flutter/src/gestures/events.dart' ('/../flutter/packages/flutter/lib/src/gestures/events.dart'). super.addAllowedPointer(event);

**Device Information [optional]:** Android

**Additional context**

Error Caused due to the plugin : `photo_view: 0.11.1 `

**Are you working on this issue? (Yes/No)** No

### PR DESCRIPTION
**Pull request by manjotsidhu**
Fixes #135, Fixes #138

## Migrate to Flutter 2.5

Changes :
- Breaking Changes in ThemeData: https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties#accentcolorbrightness.
- Update `photo_view` to `0.12.0` for flutter 2.5 compatibility.
- Revert #134 due to failing UI Tests.
- Update kotlin version

### CODE REVIEW

**Comment by RakeshSeervi** [comment]
the iOS build failed for me locally...

I followed this [answer](https://stackoverflow.com/a/64474526) and the build was successful.

Does this help?

**Comment by manjotsidhu** [comment]
> the iOS build failed for me locally...
> 
> I followed this [answer](https://stackoverflow.com/a/64474526) and the build was successful.
> 
> Does this help?

Thanks @RakeshSeervi, could you make a Pull Request to `fix-135` branch with the changes that were made by the steps you followed ?

**Comment by RakeshSeervi** [comment]
sure, sending it right away

File: lib/ib_theme.dart [hunk 1]
```diff
@@ -5,10 +5,12 @@ class IbTheme {
 
   static ThemeData getThemeData(context) {
     return Theme.of(context).copyWith(
+      brightness: Theme.of(context).brightness,
       primaryIconTheme: Theme.of(context).primaryIconTheme.copyWith(
             color: Colors.white,
           ),
-      accentColor: IbTheme.primaryColor,
+      appBarTheme: AppBarTheme(
+          foregroundColor: Colors.white, backgroundColor: IbTheme.primaryColor),
```
Comment (Nitish145) [inline lib/ib_theme.dart:13]: Don't miss trailing commas please, leads to better formatting.

**Review by Nitish145 — APPROVED** [review]

File: lib/ib_theme.dart (hunk 1 above)
Comment (manjotsidhu) [inline lib/ib_theme.dart:13]:
```suggestion
          foregroundColor: Colors.white, 
          backgroundColor: IbTheme.primaryColor,
      ),
```

**Pull request merged**